### PR TITLE
[3629] Reactify Targeting dialog

### PR DESCRIPTION
**Pull request by vestrada**
https://github.com/craftercms/craftercms/issues/3629 - Reactify Targeting dialog

### CODE REVIEW

File: ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx [hunk 1]
```diff
@@ -10,12 +13,76 @@ const translations = defineMessages({
...
+        },
+        () => {
+          setItems(
+            []
+          );
+        }
+      );
+  }
+
+  getConfigurationFile();
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx]: This is a big react no-no. Side effects should be done inside useEffect never in the body of the render function.

File: ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx [hunk 2]
```diff
@@ -10,12 +13,76 @@ const translations = defineMessages({
...
+export default function AudiencesPanel(props: AudiencesPanelProps) {
+
+  const classes = useStyles({});
+  const [items, setItems] = useState();
+  const {
+    siteId
+  } = props;
+
+  function getConfigurationFile() {
+    get(`/studio/api/1/services/api/1/site/get-configuration.json?site=editorial&path=/targeting/targeting-config.xml`)
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx]: See `services/configuration`

Create a `getAudienceTargetingConfig` method. This method will use `getDOM` to retrieve the config and then deserialize it on to a JS object with the appropriate semantics.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/services/configuration.ts [hunk 3]
```diff
@@ -95,6 +108,43 @@ export function getPreviewToolsConfig(site: string): Observable<PreviewToolsConf
...
+    map((content) => {
+      try {
+        return JSON.parse(content);
+      } catch (e) {
+        // Not JSON, assuming XML
+        let audiencesPanelConfig: AudiencesPanelConfig = { properties: null };
+        const xml = fromString(content);
+        audiencesPanelConfig.properties = Array.from(xml.querySelectorAll('property')).map((elem) => {
+
+          const name = elem.getElementsByTagName('name')[0].innerHTML,
```
Comment (rart) [inline ui/app/src/services/configuration.ts]: 1. use querySelector
2. use utils/xml/getInnerHtml 
3. use utils/xml/getInnerHtmlNumber for numeric values

File: ui/app/src/services/configuration.ts [hunk 4]
```diff
@@ -95,6 +108,43 @@ export function getPreviewToolsConfig(site: string): Observable<PreviewToolsConf
   );
 }
 
+export function getAudiencesPanelConfig(site: string): Observable<AudiencesPanelConfig> {
```
Comment (rart) [inline ui/app/src/services/configuration.ts:115]: Notice the "region" comments on 35 and 189. Create a region for your module within this file outside of the Preview config region.

**Comment by sumerjabri** [comment]
@rart is this ready?

**Comment by rart** [comment]
Content will be merged as part of https://github.com/craftercms/studio-ui/pull/1507

**Pull request closed without merging**